### DISCUSSION TITLE
Solving CORS problems with Nginx

### DISCUSSION THREAD

**Discussion by madanyang** [plausible/analytics#2053] [Self-Hosted Support]
Hi, 
I wanted to share my solution to CORS Settings with nginx as the proxy server. It can be applied to as many domains as you want. Probably it is not the best solution yet this has enabled me to use it on three different domains with no problems at all.

In the nginx site configuration file, I use the /etc/nginx/sites-available/default before the server definition add the following adjusting for your domains.
```
map $http_origin $cors_header {
default "";
~*^https://([^/]+\.)*(domainA|domainB|domainC)\.(com|net|org)$ $http_origin;
}
```
Then in the location settings
```
location / {
add_header          Access-Control-Allow-Origin  "$cors_header" always;
... Your other settings
...
}
```
This should do the trick for serving plausible for multiple domains with no CORS problems. Suggestions and corrections are much appreciated

**Comment by krchavdarov** [comment]
Hi @madanyang thanks for solution, but for me not work.
Now i have **The 'Access-Control-Allow-Origin' header contains multiple values '*, https://az-moga.bg', but only one is allowed.**

![Screenshot_32](https://user-images.githubusercontent.com/25974442/180724679-92341e82-695e-46d7-be00-6e1d12e291c3.png)

do you have some ideas?

**Reply by madanyang** [reply]
1. Do you have the settings as suggested above in your configuration ? The first part should come before 
```
server{ 
....
 }
```
and should contain only the domains no subdomains
2. Did you check with nginx -t and then restarted nginx after the changes

**Reply by krchavdarov** [reply]
This is my default nginx config
![Screenshot_34](https://user-images.githubusercontent.com/25974442/180731841-19beec4b-73a6-4de4-a74d-164e80931dad.png)
and this is plausible nginx config 
![Screenshot_35](https://user-images.githubusercontent.com/25974442/180731894-9a43fbad-2630-4f8c-b219-c8d1acc22234.png)

**Reply by madanyang** [reply]
Try with add_header in the plausible part
```
add_header          Access-Control-Allow-Origin  "$cors_header" always;
```

so it should look like
```
server {
server_name stat.az-moga.bg;
location / {
add_header          Access-Control-Allow-Origin  "$cors_header" always;
proxy_pass  http://localhost:8000;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}
```
Other than that no idea why it is not working for you

**Comment by krchavdarov** [comment]
Now get: CORS policy: The 'Access-Control-Allow-Origin' header contains multiple values '*, https://az-moga.bg', but only one is allowed.


![Screenshot_36](https://user-images.githubusercontent.com/25974442/180736075-6c354fc1-fc01-46b4-8e26-8f286f045caf.png)
